### PR TITLE
[pallet-revive] adjust fee dry-run calculation

### PR DESCRIPTION
**Pull request by pgherveou**
- Fix bare_eth_transact so that it estimate more precisely the transaction fee
- Add some context to the build.rs to make it easier to troubleshoot errors
- Add TransactionBuilder for the RPC tests.
- Improve error message, proxy rpc error from the node and handle reverted error message
- Add logs in ReceiptInfo

### CODE REVIEW

File: substrate/frame/revive/rpc/src/lib.rs [hunk 1]
```diff
@@ -252,12 +252,7 @@ impl EthRpcServer for EthRpcServerImpl {
 			.client
 			.dry_run(&transaction, block.unwrap_or_else(|| BlockTag::Latest.into()))
 			.await?;
-		let output = dry_run.result.map_err(|err| {
-			log::debug!(target: LOG_TARGET, "Dry run failed: {err:?}");
-			ClientError::DryRunFailed
-		})?;
-
-		Ok(output.into())
+		Ok(dry_run.result.into())
```
Comment (athei) [inline substrate/frame/revive/rpc/src/lib.rs:255]: I can't find this `From` implementation. Would be interested in how the error is converted.

Comment (pgherveou) [reply substrate/frame/revive/rpc/src/lib.rs:255]: Bytes is just wrapper for `Vec<u8>`

Comment (pgherveou) [reply substrate/frame/revive/rpc/src/lib.rs:255]: https://github.com/paritytech/polkadot-sdk/blob/pg/revive-fixes/substrate/frame/revive/src/evm/api/byte.rs?plain=1#L111-L115

**Review by xermicus — APPROVED** [review]

**Pull request merged**